### ISSUE TITLE
Error using code analysis in spyder

### ISSUE THREAD

**Issue by t-reents** [spyder-ide/spyder#14974]
### What steps will reproduce the problem?

I just tried to run the code analysis in spyder. Moreover I reinstalled it, but the error still occurs.

### Traceback
```python-traceback
  File "C:\Users\Timo\anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Timo\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\__main__.py", line 18, in <module>
    pylint.run_pylint()
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\__init__.py", line 22, in run_pylint
    PylintRun(sys.argv[1:])
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\lint\run.py", line 349, in __init__
    linter.check(args)
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\lint\pylinter.py", line 862, in check
    self._check_files(
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\lint\pylinter.py", line 895, in _check_files
    for name, filepath, modname in file_descrs:
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\lint\pylinter.py", line 952, in _iterate_file_descrs
    for descr in self._expand_files(files_or_modules):
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\lint\pylinter.py", line 960, in _expand_files
    result, errors = utils.expand_modules(
  File "C:\Users\Timo\anaconda3\lib\site-packages\pylint\utils\utils.py", line 159, in expand_modules
    modutils.modpath_from_file(something, path=additional_search_path)
TypeError: modpath_from_file() got an unexpected keyword argument 'path'

WARNING:spyder.widgets.github.backend:Failed to send bug report on Github. response={'code': 404, 'json': {'message': 'Not Found', 'documentation_url': 'https://docs.github.com/rest/reference/issues#create-an-issue'}}
```

## Versions

* Spyder version: 4.2.3 
* Python version: 3.8.5
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.21.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.3.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
paramiko >=2.4.0                :  2.7.2 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.8.1 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.1 (OK)
qtconsole >=5.0.1               :  5.0.2 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.4 (OK)
setuptools >=39.0.0             :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                  :  3.5.2 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  0.29.22 (OK)
matplotlib >=2.0.0              :  3.3.4 (OK)
numpy >=1.7                     :  1.20.1 (OK)
pandas >=1.1.1                  :  1.2.3 (OK)
scipy >=0.17.0                  :  1.6.1 (OK)
sympy >=0.7.3                   :  1.7.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @t-reents, thanks for reporting. As far as I can, the error you posted above is internal to Pylint and not related to Spyder. Please update to a more recent Pylint version when it's available in Anaconda (you're using Pylint 2.6.0 but the latest one is 2.7.2).

To do that, please open the Anaconda Prompt and run there

    conda update anaconda
    conda update pylint

If that doesn't work, please be patient. Unfortunately Anaconda takes months now to update to new package versions.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
